internal/database/repos_perm.go

```diff
@@ -40,19 +52,40 @@ func AuthzQueryConds(ctx context.Context, db DB) (*sqlf.Query, error) {
 	//
 	// 🚨 SECURITY: internal requests bypass authz provider permissions checks,
 	// so correctness is important here.
-	bypassAuthz := a.IsInternal() || (authzAllowByDefault && len(authzProviders) == 0)
-	if !bypassAuthz && a.IsAuthenticated() {
+	if a.IsInternal() {
+		params.BypassAuthz = true
+		params.BypassAuthzReason = BypassAuthzReasonIsInternal
+	} else if authzAllowByDefault && len(authzProviders) == 0 {
+		params.BypassAuthz = true
+		params.BypassAuthzReason = BypassAuthzReasonNoAuthzProvider
+	} else if a.IsAuthenticated() {
 		currentUser, err := db.Users().GetByCurrentAuthUser(ctx)
 		if err != nil {
 			return nil, err
 		}
-		authenticatedUserID = currentUser.ID
-		bypassAuthz = currentUser.SiteAdmin && !conf.Get().AuthzEnforceForSiteAdmins
+		params.AuthenticatedUserID = currentUser.ID
+		params.BypassAuthz = currentUser.SiteAdmin && !conf.Get().AuthzEnforceForSiteAdmins
+
+		if params.BypassAuthz {
+			params.BypassAuthzReason = BypassAuthzReasonSiteAdmin
+		}
+	}
+
+	return params, err
+}
+
+// AuthzQueryConds returns a query clause for enforcing repository permissions.
+// It uses `repo` as the table name to filter out repository IDs and should be
+// used as an AND condition in a complete SQL query.
+func AuthzQueryConds(ctx context.Context, db DB) (*sqlf.Query, error) {
+	params, err := GetAuthzQueryParameters(ctx, db)
+	if err != nil {
+		return nil, err
 	}
 
-	q := authzQuery(bypassAuthz,
-		usePermissionsUserMapping,
-		authenticatedUserID,
+	q := authzQuery(params.BypassAuthz,
+		params.UsePermissionsUserMapping,
+		params.AuthenticatedUserID,
```

Comment:
I think since we already have it in object, let's simplify?

```suggestion
	q := authzQuery(params,
```